cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/si.hpp

```diff
@@ -102,9 +100,18 @@ subgraph_isomorphism::graph_matching_result si_call_kernel(
         pattern.set_vertex_attribute(p_vertex_count, vv_p);
     }
 
-    solution<Cpu> results = si<Cpu>(pattern, target, si_kind, alloc_ptr);
+    inner_alloc local_allocator(alloc_ptr);
+    auto sorted_pattern_vertex = local_allocator.make_shared_memory<std::int64_t>(p_vertex_count);
+    std::int64_t* sorted_pattern_vertex_array = sorted_pattern_vertex.get();
+    if (sorted_pattern_vertex_array == nullptr) {
+        throw oneapi::dal::host_bad_alloc();
+    }
```

Comment:
Why not to make this check in `make_shared_memory`? I saw usage of `make_shared_memory` not followed by this check.

Reply:
@rlnx, thank you for code commenting. I agree with you, all these issues will be fixed.